react-app/src/components/SaveOutputForm.js

```diff
@@ -0,0 +1,179 @@
+import React, { useState } from 'react';
+import { Alert, Button, Col, Form, Row } from 'react-bootstrap';
+import getLabel from './patientUtils';
+
+function SaveOutputForm(props) {
+  const [saveLogs, setSaveLogs] = useState(false);
+
+  const [showSavedAlert, setShowSavedAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showNoFilesAlert, setShowNoFilesAlert] = useState(false);
+
+  let defaultWhichFiles = {};
+  props.extractedData.forEach((bundle, i) => {
+    const label = getLabel(bundle, i);
+    defaultWhichFiles[label] = true;
+  });
+  const [whichFiles, setWhichFiles] = useState({ ...defaultWhichFiles });
+
+  function toggleSaveLogs() {
+    setSaveLogs(!saveLogs);
+  }
+
+  function isAllSelected() {
+    let isAll = true;
+    props.extractedData.forEach((bundle, i) => {
+      const label = getLabel(bundle, i);
+      if (whichFiles[label] === false) {
+        isAll = false;
+      }
+    });
+    return isAll;
+  }
+
+  function togglePatientCheckbox(e) {
+    setWhichFiles({ ...whichFiles, [e.target.id]: e.target.checked });
+  }
+
+  function getPatientCheckboxes() {
+    const list = props.extractedData.map((bundle, i) => {
+      const label = getLabel(bundle, i);
+      return (
+        <Form.Check
+          type="checkbox"
+          label={label}
+          key={label}
+          id={label}
+          checked={whichFiles[label]}
+          onChange={togglePatientCheckbox}
+        />
+      );
+    });
+    return list;
+  }
+
+  function toggleSelectAll() {
+    // modify which files will be saved first, so that there's no issues with state update delays
+    if (!isAllSelected()) {
+      props.extractedData.forEach((bundle, i) => {
+        const label = getLabel(bundle, i);
+        defaultWhichFiles = {
+          ...defaultWhichFiles,
+          [label]: true,
+        };
+      });
```

Comment:
Same comment as earlier regarding the object spread operator here.

Reply:
I have modified the format of this assignment